File: apps/traductor-pro/server/src/routes/documents.ts
import { Router } from 'express';
import multer from 'multer';
import pdfParse from 'pdf-parse';
import mammoth from 'mammoth';

const router = Router();
const upload = multer({ storage: multer.memoryStorage(), limits: { fileSize: 10 * 1024 * 1024 } });

router.post('/extract-text', upload.single('file'), async (req, res) => {
  const file = req.file;
  if (!file) return res.status(400).json({ error: 'No se recibió ningún archivo' });

  try {
    let texto = '';

    if (file.mimetype === 'application/pdf' || file.originalname.endsWith('.pdf')) {
      const data = await pdfParse(file.buffer);
      texto = data.text;
    } else if (
      file.mimetype === 'application/vnd.openxmlformats-officedocument.wordprocessingml.document' ||
      file.originalname.endsWith('.docx')
    ) {
      const result = await mammoth.extractRawText({ buffer: file.buffer });
      texto = result.value;
    } else {
      texto = file.buffer.toString('utf-8');
    }

    return res.json({ texto: texto.trim() });
  } catch (err: any) {
    console.error(err);
    return res.status(500).json({ error: 'Error extrayendo texto', details: err?.message });
  }
});

export default router;

File: apps/traductor-pro/server-fastapi/app/routes/documents.py
import logging
from fastapi import APIRouter, HTTPException, UploadFile, File

logger = logging.getLogger("traductor")

router = APIRouter()


@router.post("/documents/extract-text")
async def extract_text(file: UploadFile = File(...)):
    if not file:
        raise HTTPException(400, "No se recibió ningún archivo")

    contents = await file.read()
    if len(contents) > 10 * 1024 * 1024:
        raise HTTPException(400, "Archivo demasiado grande (máx 10MB)")

    try:
        texto = ""

        if file.content_type == "application/pdf" or (file.filename and file.filename.endswith(".pdf")):
            from pypdf import PdfReader
            import io
            reader = PdfReader(io.BytesIO(contents))
            texto = "\n".join(page.extract_text() or "" for page in reader.pages)
        elif file.content_type == "application/vnd.openxmlformats-officedocument.wordprocessingml.document" or (
            file.filename and file.filename.endswith(".docx")
        ):
            from docx import Document
            import io
            doc = Document(io.BytesIO(contents))
            texto = "\n".join(p.text for p in doc.paragraphs)
        else:
            texto = contents.decode("utf-8")

        return {"texto": texto.strip()}
    except Exception as e:
        logger.error(f"Error extrayendo texto: {e}")
        raise HTTPException(500, f"Error extrayendo texto: {e}")
